src/CFamily/CMake/VsDevCmdEnvironmentProvider.cs

```diff
@@ -0,0 +1,231 @@
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Abstractions;
+using System.Text;
+using Microsoft.VisualStudio;
+using SonarLint.VisualStudio.CFamily.SystemAbstractions;
+using SonarLint.VisualStudio.Infrastructure.VS;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Integration.Helpers;
+
+namespace SonarLint.VisualStudio.CFamily.CMake
+{
+    internal interface IVsDevCmdEnvironmentProvider
+    {
+        /// <summary>
+        /// Returns the environment settings set by VsDevCmd.bat
+        /// </summary>
+        /// <param name="scriptParams">Any additional parameters to pass to the batch file. Can be null.</param>
+        IReadOnlyDictionary<string, string> Get(string scriptParams);
+    }
+
+    internal class VsDevCmdEnvironmentProvider : IVsDevCmdEnvironmentProvider
+    {
+        /// <summary>
+        /// Relative path from the VS installation root to the batch file
+        /// </summary>
+        /// <remarks>This path works for VS2015+</remarks>
+        private const string RelativePathToBatchFile = "Common7\\Tools\\VsDevCmd.bat";
+
+        private const int SCRIPT_TIMEOUT_MS = 4000;
+
+        private readonly IVsInfoService vsInfoService;
+        private readonly ILogger logger;
+
+        // Interfaces for testing
+        private readonly IProcessFactory processFactory;
+        private readonly IFileSystem fileSystem;
+
+
+        public VsDevCmdEnvironmentProvider(IVsInfoService vsInfoService, ILogger logger)
+            : this(vsInfoService, logger, new ProcessFactory(), new FileSystem())
+        {
+            this.vsInfoService = vsInfoService;
+            this.logger = logger;
+        }
+
+        internal /* for testing */ VsDevCmdEnvironmentProvider(IVsInfoService vsInfoService, ILogger logger,
+            IProcessFactory processFactory, IFileSystem fileSystem)
+        {
+            this.vsInfoService = vsInfoService;
+            this.logger = logger;
+            this.processFactory = processFactory;
+            this.fileSystem = fileSystem;
+        }
+
+        public IReadOnlyDictionary<string, string> Get(string scriptParams)
```

Comment:
Overall the implementation is similar to the VS version, but will more logging and interfaces to make it testable.